src/components/select/select.tsx

```diff
@@ -1,21 +1,57 @@
 import { SELECT_OPTIONS } from '@constants';
-import React, { FC } from 'react';
-import shortid from 'shortid';
-import { SelectMain, SelectTitle, SelectWrapper } from './styles';
+import React, { FC, useState } from 'react';
+import {
+  SelectInput,
+  SelectLabel,
+  SelectMain,
+  SelectOver,
+  SelectTitle,
+  SelectWrapper,
+} from './styles';
 import { SelectProps } from './types';
 
-export const Select: FC<SelectProps> = ({ onChange, name }) => {
-  const options = SELECT_OPTIONS.map((option) => (
-    <option key={shortid.generate()}>{option}</option>
-  ));
+export const Select: FC<SelectProps> = ({ selected, onChange }) => {
+  const [isOpen, setOpen] = useState(false);
+
+  const handleOnSelected = (selectedOption) => {
+    if (selected.includes(selectedOption)) {
+      const newSelected = selected.filter(
+        (option) => option !== selectedOption,
+      );
+      onChange(newSelected);
+    } else {
+      const newSelected = [...selected, selectedOption];
+      onChange(newSelected);
+    }
+  };
+
+  const options = SELECT_OPTIONS.map((option) => {
```

Comment:
It is better to move it to separate components since here you moving JSX rendering to a separate variable. Or insert it int jsx